app/code/Magento/Bundle/Test/Unit/Observer/SetAttributeTabBlockObserverTest.php

```diff
@@ -0,0 +1,112 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\Bundle\Test\Unit\Observer;
+
+use Magento\Bundle\Block\Adminhtml\Catalog\Product\Edit\Tab\Attributes;
+use Magento\Bundle\Observer\SetAttributeTabBlockObserver;
+use Magento\Catalog\Helper\Catalog;
+use Magento\Catalog\Model\Product;
+use Magento\Catalog\Model\Product\Type;
+use Magento\Framework\Event;
+use Magento\Framework\Event\Observer;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * Class SetAttributeTabBlockObserverTest
+ *
+ * Test setting attribute tab block for bundle products
+ */
+class SetAttributeTabBlockObserverTest extends TestCase
+{
+    /**
+     * @var SetAttributeTabBlockObserver
+     */
+    private $observer;
+
+    /**
+     * @var Catalog|MockObject
+     */
+    private $helperCatalogMock;
+
+    /**
+     * @var Observer|MockObject
+     */
+    private $observerMock;
+
+    /**
+     * @var Event|MockObject
+     */
+    private $eventMock;
+
+    /**
+     * @var Product|MockObject
+     */
+    private $productMock;
+
+    /**
+     * Set Up
+     */
+    public function setUp()
+    {
+        $this->helperCatalogMock = $this->createMock(Catalog::class);
+        $this->observerMock = $this->createMock(Observer::class);
+        $this->eventMock = $this->getMockBuilder(Event::class)
+            ->disableOriginalConstructor()
+            ->setMethods(['getProduct'])
+            ->getMock();
+        $this->productMock = $this->getMockBuilder(Product::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $this->observer = new SetAttributeTabBlockObserver(
```

Comment:
Use ObjectManager helper for Unit Tests.

Reply:
Done 👌